core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.spec.ts

```diff
@@ -72,6 +72,23 @@ describe('EmaContentletToolsComponent', () => {
             expect(spectator.query(byTestId('drag-image'))).toHaveText('test');
         });
 
+        it('should close menus when contentlet @input was changed', () => {
+            const spyHideMenus = jest.spyOn(spectator.component, 'hideMenus');
```

Comment:
check for the menu's `hide()` instead ?

Reply:
We can only check the hide of one menu (If one is open and another is selected, the previous one is closed, so the .hide would be executed to undefined)

But I just added a case for one menu

Reply:
Never mock a method on the component you're testing, mock the `menu.hide` and the `menuVtl.hide`